server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -192,6 +201,83 @@ public void clear(String tenantId) {
         }
     }
 
+    @Override
+    public Trace getTraceFragment(String tenantId, String id) {
+        Span span = getSpan(tenantId, id);
+
+        Trace trace = spanToTrace(span);
+        LinkedList<Node> nodes = recursiveTraceFragment(tenantId, id);
+        nodes.push(spanToNode(span));
```

Comment:
The result of 'spanToNode(span)' should be added as a node on the trace. The list 'nodes' should then be set as the node list on the node returned from spanToNode.